src/Tabs/TabGroup.js

```diff
@@ -0,0 +1,111 @@
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+import { TabContent } from './_TabContent';
+import React, { Component } from 'react';
+
+export class TabGroup extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            selectedId: props.selectedId
+        };
+    }
+
+    static getDerivedStateFromProps(nextProps, prevState) {
+        return nextProps.selectedId !== prevState.selectedId ? { selectedId: nextProps.selectedId } : null;
+    }
+
+    // set selected tab
+    handleTabSelection = (event, id) => {
+        event.preventDefault();
+        this.setState({
+            selectedId: id
+        });
+
+        if (this.props.onTabClick) {
+            this.props.onTabClick(event, id);
+        }
+    };
+
+    // clone Tab element
+    cloneElement = (child) => {
+        return (React.cloneElement(child, {
+            onClick: this.handleTabSelection,
+            selected: this.state.selectedId === child.props.id
+        }));
+    }
+
+    // create tab list
+    renderTabs = () => {
+        return React.Children.map(this.props.children, (child) => {
+            return (
+                <li
+                    {...this.props.tabLinkProps}
+                    className='fd-tabs__item'
+                    key={child.props.id}>
+                    {this.cloneElement(child)}
+                </li>);
+        });
+    };
+
+    // create content to show below tab list
+    renderContent = () => {
+        return React.Children.map(this.props.children, (child) => {
+            return (
+                <TabContent
+                    {...this.props.tabContentProps}
+                    selected={this.state.selectedId === child.props.id}>
+                    {child.props.children}
+                </TabContent>);
+        });
+    };
+
+    render() {
+        const {
+            children,
+            className,
+            selectedId,
+            tabContentProps,
+            tabLinkProps,
+            tabGroupProps,
+            ...rest } = this.props;
+
+        // css classes to use for tab group
+        const tabGroupClasses = classnames(
+            'fd-tabs',
+            className
+        );
+        return (
+            <React.Fragment>
+                <ul {...rest}
+                    className={tabGroupClasses}
+                    role='tablist'>
+                    {this.renderTabs(children)}
+                </ul>
+                {this.renderContent(children)}
+            </React.Fragment>
+        );
+    }
+}
+TabGroup.displayName = 'TabGroup';
+
+TabGroup.defaultProps = {
+    selectedId: '1'
```

Comment:
I think this may be problematic.  This relies on the consumer setting the `id` prop on the first tab to `1`.  Maybe this should be changed to a number prop and just simply be the index of the selected tab rather than a specific `id` value.